internal/commands/vaultsecrets/secrets/open.go

```diff
@@ -165,8 +147,8 @@ func openRun(opts *OpenOpts) (err error) {
 		return nil
 	}
 
-	displayer := newDisplayer(true).OpenAppSecrets(resp.Payload.Secret).
-		SetDefaultFormat(format.Pretty).AddFields(field)
+	displayer := newDisplayer().OpenAppSecrets(resp.Payload.Secret).SetDefaultFormat(format.Pretty).
+		SetSingleSecret(resp.Payload.Secret.Type)
```

Comment:
A boolean argument to `newDisplayer()` was the indication to the displayer that it was working with a single secret or list of secrets before. With this change, I've introduced a `SetSingleSecret(type)`. I think this is more clear but could use feedback.

Reply:
When chaining `OpenAppSecrets(resp.Payload.Secret)` it gives the displayer access to the secret's type.

If the displayer internally had a `map[secretType]openSecretFormatter` it'd encapsulate that complexity for the callers and remove the need to use switch cases.

Reply:
I like this idea. Reworking it now!

Reply:
Okay I introduced an interface and removed all of the secret type switch cases in https://github.com/hashicorp/hcp/pull/129/commits/6cd82c7fef8fbe6e2edaf7d52a029ace943930ca. I was also able to simplify by deleting `SetSecretType` and the private `single` method. Those details are encapsulated now so that callers don't need to set them.